### PR TITLE
Add test for failing start function

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/unittests/instantiate_test.cpp [hunk 1]
```diff
@@ -550,3 +550,15 @@ TEST(instantiate, globals_initialized_from_imported)
     EXPECT_THROW_MESSAGE(instantiate(module_invalid2, {}, {}), instantiate_error,
         "Global can be initialized by another const global only if it's imported.");
 }
+
+TEST(execute, start_unreachable)
+{
+    Module module;
+    module.startfunc = FuncIdx{0};
+    // TODO: add type section (once enforced)
```
Comment (gumb0) [inline test/unittests/instantiate_test.cpp]:
```suggestion
    module.typesec.emplace_back(FuncType{{}, {}});
```

Comment (axic) [reply test/unittests/instantiate_test.cpp]: I actually want to keep it so that its caught when we implement that check.

Comment (gumb0) [reply test/unittests/instantiate_test.cpp]: Maybe create a separate test for this?

Comment (axic) [reply test/unittests/instantiate_test.cpp]: Would you have noticed if i didn't had the comment there? :)

Comment (gumb0) [reply test/unittests/instantiate_test.cpp]: I wouldn't. Just not sure why leave some TODOs that will make future changes more complicated. But not a big deal.

**Pull request merged**